force-app/main/default/classes/FlowControlLoopsCollections.cls

```diff
@@ -1,368 +1,480 @@
+   */
+  public static Integer getSalary(String employeeName) {
+    // Initialize a Map
+    Map<String, Integer> employeeSalary = new Map<String, Integer>();
+    // Add key-value pairs to the Map
+    // add John Doe => 50000
+    // add Jane Smith => 60000
+    // add Sam Brown => 55000
+    // add Alice Johnson => 65000
+    employeeSalary.put('John Doe', 50000);
+    employeeSalary.put('Jane Smith', 60000);
+    employeeSalary.put('Sam Brown', 55000);
+    employeeSalary.put('Alice Johnson', 65000);
+    // Get the salary of the employee
+    Integer employeeSalaryValue = employeeSalary.get(employeeName);
+    // Return the salary of the employee, or null if the employee is not found in the Map
+    if (employeeName == null) {
+      return null;
+    } else {
+      return employeeSalaryValue;
     }
-
-    /**
-     * Question 17
-     * Loop through the list of money and adds them up until it accumulates more than 40 and then stops counting.
-     * The change values are in cents and the total is to be calculated in dollars.
-     * Example: countMoney() should return 48.02
-     * @return The total value in dollars, stopped at the point when it exceeds 40 dollars.
-     */
-    public static Decimal countMoney() {
-        // The list of money in the wallet, represented in dollars and cents
-        List<Decimal> moneyInWallet = new List<Decimal>{0.50, 10, 3.84, 24.60, 9.08, 50, 4.90}; //DO NOT CHANGE
-
-        return null; // Replace null with the variable you used to store the result
+    //return employeeName; // Replace null with the variable you used to store the result
+  }
+
+  /**
+   * Question 20
+   * Create a map where the keys are the names of employees and the values are their salaries.
+   * Then, iterates over the map and checks if an employee's salary is more than 55000.
+   * If it is, add the employee's name to a list of employees that is returned by the method.
+   * Remember that a Map is a collection of key-value pairs. Key are a set of unique values, so you can't have duplicate keys. Values are a list of values, so you can have duplicate values.
+   * Resource: https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/langCon_apex_loops_for_lists.htm
+   * Example: getHighPaidEmployees() should return ['Jane Smith', 'Alice Johnson']
+   * @return A list of employee names who have a salary more than 55000.
+   */
+  public static List<String> getHighPaidEmployees() {
+    // Initialize a Map
+    Map<String, Integer> employeeSalaries = new Map<String, Integer>();
+
+    // Add key-value pairs to the Map
+    // add John Doe => 50000
+    // add Jane Smith => 60000
+    // add Sam Brown => 55000
+    // add Alice Johnson => 65000
+    employeeSalaries.put('John Doe', 50000);
+    employeeSalaries.put('Jane Smith', 60000);
+    employeeSalaries.put('Sam Brown', 55000);
+    employeeSalaries.put('Alice Johnson', 65000);
+
```

Comment:
Try using a static variable for the employeesalaries for question 20 and 19
The values are the same so we try to reduce duplicates.